src/msg/async/rdma/Infiniband.h

```diff
@@ -79,8 +79,13 @@ class Device {
   ~Device() {
     if (active_port) {
       delete active_port;
+      active_port = nullptr;
       ceph_assert(ibv_close_device(ctxt) == 0);
     }
+    if (device_attr) {
```

Comment:
no need to check for null. as `delete` a null ptr is a no-op.

Reply:
also, i think it'd be better to define `device_attr` as a plain `ibv_device_attr` instance instead of a pointer pointing to it. as `Device` is always allocated on the heap, this change does not change the behavior of the application.

Reply:
@tchaikov 
It's true that we could use a plain ibv_device_attr obj here. However, we need to change some other parts in the source which use the pointer to access its member previously.
Do we really need to change it?

Reply:
@iotcg i think it's an opportunity to improve the quality of code. i don't feel strong though if you don't want to touch it.

Reply:
Thanks. I'll refine it when I'm available.